sdk/tests/core/test_az_http.c

```diff
@@ -180,6 +180,222 @@ static void test_http_request(void** state)
     assert_return_code(
         az_http_request_append_header(&request, AZ_SPAN_FROM_STR("header"), AZ_SPAN_NULL), AZ_OK);
   }
+  { // Test adding duplicated query parameters
+    uint8_t buf[100];
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(buf, 0, sizeof(buf));
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_BUFFER(buf);
+    az_span initial_url = AZ_SPAN_FROM_STR("http://example.com");
+    az_span remainder = az_span_copy(url_span, initial_url);
+    assert_int_equal(az_span_size(remainder), 100 - az_span_size(initial_url));
```

Comment:
nit: This assert isn't really necessary. It isn't testing anything related to http, just that the span copy worked.

Reply:
yeah,, it came together when copy pasting an existing test and just wanted to leave it there :)